### ISSUE TITLE
Support for docker stack

### ISSUE THREAD

**Issue by ivanrusso** [mariotoffia/FluentDocker#78]
Is it possible to add support for docker stack commands?

**Comment by mariotoffia** [comment]
Absolutely, I'll implement [this](https://docs.docker.com/engine/reference/commandline/stack/) sometime (hopefully) this month.

Cheers,
 Mario

**Comment by mariotoffia** [comment]
**Resolved in v3.0.0.** The v3 ground-up rewrite implements (or fixes) what this issue asked for.

This 2019 feature request asked for support for `docker stack` commands, and the owner committed to implementing it. v3 implements the full `docker stack` command surface as a first-class driver component (`IStackDriver` + `DockerCliStackDriver`), covering exactly the commands documented at the linked Docker stack reference: ls, ps, deploy, rm, and services. It is registered in the Docker CLI driver pack, exposed via the kernel (`Kernel.SysCtl<IStackDriver>`), wrapped by a fluent testing resource (`SwarmStackResource`), gated by capability checks, and covered by unit, integration, and resource tests. The request is fully satisfied.

<sub>Where this lives in v3: Interface: FluentDocker/Drivers/IStackDriver.cs defines IStackDriver with ListAsync (docker stack ls), GetTasksAsync (docker stack ps), DeployAsync (docker stack deploy), RemoveAsync (docker stack rm), GetServicesAsync (docker stack services), plus StackInfo/StackTask/StackServiceInfo/StackDeployConfig/StackDeployResult/StackListFilter models. Implementation: FluentDocker/Drivers/Docker/Cli/Components/DockerCliStackDriver.cs builds the actual `stack ls/ps/deploy/rm/services` CLI args (e.g. line 121 "stack deploy" with -c/--prune/--with-registry-auth, line 154 "stack rm"). Registration: FluentDocker/Drivers/Docker/Cli/DockerCliDriverPack.cs lines 78/90/102 instantiate, initialize, and…</sub>

Closing as part of the v3 issue sweep. If your scenario still isn't covered by v3, please reopen with a v3 reproduction and we'll take another look. Thanks for the report! 🙏
